src/recipes.jl

```diff
@@ -719,24 +719,39 @@ end
 
     # create a list of shapes, where each shape is a single boxplot
     shapes = Shape[]
-    groupby = extractGroupArgs(d[:x])
-
-    for (i, glabel) in enumerate(groupby.groupLabels)
 
-        # get the edges and widths
-        y = d[:y][groupby.groupIds[i]]
-        widths, centers = violin_coords(y, trim=trim)
+    legend --> false
 
+    if length(d[:x])==1
+        widths, centers = violin_coords(d[:y], trim=trim)
         # normalize
         widths = _box_halfwidth * widths / maximum(widths)
 
         # make the violin
-        xcenter = discrete_value!(d[:subplot][:xaxis], glabel)[1]
+        xcenter = discrete_value!(d[:subplot][:xaxis], d[:x][1])[1]
```

Comment:
No probs, take your time.  It works as far as I need it working at the moment and that publication is still a long way off...